ldp/nn/handlers/chunking.py

```diff
@@ -0,0 +1,169 @@
+        real_outputs = [
+            output
+            for output, is_dummy in zip(outputs_list, dummy_chunk_flags, strict=True)
+            if not is_dummy
+        ]
+
+        if isinstance(real_outputs[0], torch.Tensor):
+            # If we received a list of tensors, concat along the batch dimension
+            # TODO: consider how/if to handle differently shaped tensors
+            return torch.cat(real_outputs, dim=0)
+
+        if isinstance(real_outputs[0], GenerateDecoderOnlyOutput):
+            sequences = []
+            for output in real_outputs:
+                sequences.extend(output.sequences)
+            scores: list[list[torch.Tensor | None]] | None = None
+            if real_outputs[0].scores is not None:
+                assert all(output.scores is not None for output in real_outputs)
+                # `output.scores` is a tuple (one el per decoded token), with each tensor of shape
+                # (batch_size, vocab_size).
+                # To get the scores for the `i`-th batch element, we'd do `[score[i] for score in output.scores]`.
+                # `output.scores` will have different lengths across workers, so we cannot simply concatenate them.
+                # Adding dummy scores is error-prone and loses info. Instead, we create a merged `scores` with the
+                # same access semantics. `score[i]` will be a tensor if it exists for batch element `i` and `None`
+                # otherwise.
+                scores = []
+                max_output_len = max(len(output.scores) for output in real_outputs)
+                for i in range(max_output_len):
+                    scores_step_i: list[torch.Tensor | None] = []
+                    for output in real_outputs:
+                        if i < len(output.scores):
+                            scores_step_i.extend(list(output.scores[i]))
+                        else:
+                            # Add bsz Nones for this worker, since it did not reach token `i`.
+                            scores_step_i.extend([None] * output.sequences.shape[0])
+
+                    scores.append(scores_step_i)
+
+            return GenerateDecoderOnlyOutput(
+                sequences=sequences,
+                scores=scores,
+            )
+
+        raise ValueError("Unsupported output type")
+
+    def _split_value(self, value):
+        """
+        Splits a single value into chunks, adding a dummy value if necessary.
+
+        Right now, only torch.Tensor values are split. Non-tensor values are replicated.
+        """
+        if isinstance(value, torch.Tensor):
+            chunks = list(torch.chunk(value, self.num_chunks, dim=0))
+            dummy_chunk_flags = []
+            for i in range(self.num_chunks):
+                if i >= len(chunks):
+                    # Chunk 0 will always exist, and we need only a batch of one to activate the model
+                    chunks.append(torch.full_like(chunks[0][:1], self.dummy_value))
```

Comment:
Can you document the `[:1]`?

Reply:
# Chunk 0 will always exist, and we need only a batch of one to activate the model

Maybe add to that existing comment above "therefore [:1] is used"